packages/@aws-cdk/aws-redshift/lib/private/database-query-provider/table.ts

```diff
@@ -88,8 +88,11 @@ async function updateTable(
   }
 
   const oldTableColumns = oldResourceProperties.tableColumns;
-  if (!oldTableColumns.every(oldColumn => tableColumns.some(column => column.name === oldColumn.name && column.dataType === oldColumn.dataType))) {
-    return createTable(tableNamePrefix, tableNameSuffix, tableColumns, tableAndClusterProps);
+  const columnDeletions = oldTableColumns.filter(oldColumn => (
+    tableColumns.every(column => oldColumn.name !== column.name)
```

Comment:
I saw the previous code as a check for whether or not to create a new table when a column was deleted. What my change implements is to filter out the columns and store them in a new array. This will contain the columns to delete. I'm not sure how the `createTable` function works, but I'm guessing it would create a whole new table, which caused the issue that was mentioned in the description.

Reply:
ah, I see. This is correct then.